### DISCUSSION TITLE
enhancement: Lag when using over high-latency connections

### DISCUSSION THREAD

**Discussion by Fusseldieb** [open-webui/open-webui#23691] [Issues]
**Is your feature request related to a problem? Please describe.**
It's more of a UX request.

**Describe the solution you'd like**
When using open-webui over high-latency connections (eg. 300-350ms), certain page actions are incredibly delayed giving the user the impression that he hasn't clicked a button or submitted certain actions.
This is most noticeable when sending a text to a model, where the text _remains_ in the textbox even after it's being sent, and will only clear once the model answers. This becomes even worse when a image is chosen.
It's also noticeable when deleting a chat. Clicking the "yes" check to delete a chat gives no immediate answer until the server responds, which makes me sometimes wonder if I really clicked it. A simple spinner would solve this.
Furthermore, it's also noticeable when the page initially loads. The user is presented with a completely blank page and the "loading" spinner from the browser itself ceases, giving the impression that the page really has loaded, and is blank. However, several seconds later, the content actually loads. Again, a simple spinner hardcoded in HTML/CSS would solve this.

**Describe alternatives you've considered**
\-

**Additional context**
My only server that runs 24/7 is (unfortunately) pretty far away, and thus I notice this all the time.

**Comment by tjbck** [comment]
Related: #1550 Could you try our dev branch and check if anything's changed for you?

**Comment by Fusseldieb** [comment]
I currently pulled and built from dev, as you requested, and the same issues persist. Nothing's changed. 

I have since relocated the UI to another server closer to me. However, I saw that the _exact same_ behaviour can be archieved using DevTools/Network and creating a new "High Latency" profile (350ms).

<details>
  <summary>Images (click to expand)</summary>

![image](https://github.com/open-webui/open-webui/assets/4715129/20206811-dbab-49dc-8cd9-b021edd66c99)

![image](https://github.com/open-webui/open-webui/assets/4715129/fed96b12-4eeb-4043-854e-848d2d58c87e)
</details>
